File: KidozSDKSampleApp/ViewController.m
//
//  ViewController.m
//  Kidoz_iOS_SDK_Sample_App
//
//  Created by Lev Firer on 12/02/2018.
//  Copyright © 2018 Kidoz. All rights reserved.
//

#import "ViewController.h"

#import <AudioToolbox/AudioToolbox.h>
#import <AVFoundation/AVFoundation.h>

@interface ViewController ()
@property (strong, nonatomic) AVAudioPlayer *player;
@end


@implementation ViewController
@synthesize logText;
@synthesize player;

-(void)setBorder{
    [[self.logText layer] setBorderColor:[[UIColor orangeColor] CGColor]];
    [[self.logText layer] setBorderWidth:2.3];
    [[self.logText layer] setCornerRadius:15];
}

-(void)logOut:(NSString*)message withUITextView:(UITextView*)textView withTimestamp:(NSString*)timestamp{
    
    textView.text = [textView.text stringByAppendingString:timestamp];
    textView.text = [textView.text stringByAppendingString:@" - "];
    
    textView.text = [textView.text stringByAppendingString:message];
    textView.text = [textView.text stringByAppendingString:@"\n"];
    
    NSRange bottom = NSMakeRange(logText.text.length -1, 1);
    [textView scrollRangeToVisible:bottom];
    
}

/*     Kidoz SDK     */

-(void)onInitSuccess{
    [self logOut:@"SDK Initialized" withUITextView:logText withTimestamp:[self getTimestamp]];


                            
    [self logOut:@"Interstitial initializing..." withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance] initializeInterstitialWithDelegate:self];
    
    [self logOut:@"Rewarded initializing..." withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance] initializeRewardedWithDelegate:self];
    
    [self logOut:@"Banner initializing..." withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance]initializeBannerWithDelegate:self withViewController:self];
    [[KidozSDK instance]setBannerPosition:BOTTOM_CENTER];

}

-(void)onInitError:(NSString *)error{
    [self logOut:@"SDK Init Error" withUITextView:logText withTimestamp:[self getTimestamp]];
}

/*     Banner     */

- (IBAction)loadBanner:(id)sender {
    if([[KidozSDK instance]isBannerInitialized]){
        [self logOut:@"Load Banner" withUITextView:logText withTimestamp:[self getTimestamp]];
        [[KidozSDK instance]loadBanner];
    }
}


- (IBAction)showBanner:(id)sender {
    [self logOut:@"Show Banner" withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance]showBanner];
    
}

- (IBAction)hideBanner:(id)sender {
     [[KidozSDK instance]hideBanner];
}

- (void)bannerDidClose {
    [self logOut:@"Banner Did Close" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerDidInitialize {
    [self logOut:@"Banner Did Initialize" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerDidOpen {
    [self logOut:@"Banner Did Open" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerDidReciveError:(NSString *)errorMessage {
    [self logOut:errorMessage withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerIsReady {
    [self logOut:@"Banner Is Ready" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerLeftApplication {
    [self logOut:@"banner Left Application" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerLoadFailed {
    [self logOut:@"Banner Load Failed" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)bannerReturnedWithNoOffers {
    [self logOut:@"Banner Returned With No Offers" withUITextView:logText withTimestamp:[self getTimestamp]];
}


/*      --------       */


/*     Interstitial     */



- (IBAction)loadInterstitial:(id)sender {
   
    if([[KidozSDK instance]isInterstitialInitialized]){
         [self logOut:@"Load Interstitial" withUITextView:logText withTimestamp:[self getTimestamp]];
        [[KidozSDK instance]loadInterstitial];
    }

}

- (IBAction)showInterstitial:(id)sender {
    [self logOut:@"Show Interstitial" withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance]showInterstitial];
}



-(void)interstitialDidInitialize{
    [self logOut:@"Interstitial Initialized" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)interstitialDidClose{
    if(player!= nil)
        [player setVolume:1.0];
    
    [self logOut:@"Interstitial Closed" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)interstitialDidOpen{
    if(player!= nil)
        [player setVolume:0.0];
    
    [self logOut:@"Interstitial Opened" withUITextView:logText withTimestamp:[self getTimestamp]];
}


-(void)interstitialIsReady{
    [self logOut:@"Interstitial Ready" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)interstitialDidPause{
    [self logOut:@"Interstitial Did Pause" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)interstitialDidResume{
    [self logOut:@"Interstitial Did Resume" withUITextView:logText withTimestamp:[self getTimestamp]];
}



-(void)interstitialReturnedWithNoOffers{
    [self logOut:@"Interstitial Returned With No Offers" withUITextView:logText withTimestamp:[self getTimestamp]];
}


-(void)interstitialDidReciveError:(NSString*)errorMessage{
    [self logOut:errorMessage withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)interstitialLoadFailed {
    [self logOut:@"interstitial Load Failed" withUITextView:logText withTimestamp:[self getTimestamp]];

}

- (void)interstitialLeftApplication {
    [self logOut:@"interstitial Left Application" withUITextView:logText withTimestamp:[self getTimestamp]];

}

/*      --------       */

/*      Rewarded       */
- (IBAction)loadRewarded:(id)sender {
    if([[KidozSDK instance]isRewardedInitialized]){
        [self logOut:@"Load Rewarded" withUITextView:logText withTimestamp:[self getTimestamp]];
        [[KidozSDK instance]loadRewarded];
    }
}

- (IBAction)showRewarded:(id)sender {
    [self logOut:@"Show Rewarded" withUITextView:logText withTimestamp:[self getTimestamp]];
    [[KidozSDK instance]showRewarded];
}

-(void)rewardedDidInitialize{
    [self logOut:@"Rewarded Initialized" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedDidClose{
    if(player!= nil)
        [player setVolume:1.0];
    
    [self logOut:@"Rewarded Closed" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedDidOpen{
    if(player!= nil)
        [player setVolume:0.0];
    
    [self logOut:@"Rewarded Opened" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedIsReady{
    [self logOut:@"Rewarded Ready" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedReturnedWithNoOffers{
    [self logOut:@"Rewarded Returned With No Offers" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedDidPause{
    [self logOut:@"Rewarded Did Pause" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedDidResume{
    [self logOut:@"Rewarded Did Resume" withUITextView:logText withTimestamp:[self getTimestamp]];
}


-(void)rewardedDidReciveError:(NSString*)errorMessage{
    [self logOut:errorMessage withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardReceived{
    [self logOut:@"Reward Received" withUITextView:logText withTimestamp:[self getTimestamp]];
}

-(void)rewardedStarted{
    [self logOut:@"Rewarded Started" withUITextView:logText withTimestamp:[self getTimestamp]];
}

- (void)rewardedLoadFailed {
    [self logOut:@"Rrewarded Load Failed" withUITextView:logText withTimestamp:[self getTimestamp]];

}

- (void)rewardedLeftApplication {
    [self logOut:@"rewarded Left Application" withUITextView:logText withTimestamp:[self getTimestamp]];

}



/*      --------       */

-(NSString*)getTimestamp{    
    NSDateFormatter *objDateformat = [[NSDateFormatter alloc] init];
    [objDateformat setDateFormat:@"HHmmssSSS"];
    //[objDateformat setDateFormat:@"yyyyMMdd'T'HHmmss.SSS"];
    NSString *timestamp = [objDateformat stringFromDate:[NSDate date]];
    
    return timestamp;
}


/*
-(BOOL)shouldAutorotate{
    return NO;
}


-(UIInterfaceOrientationMask)supportedInterfaceOrientations{
    return UIInterfaceOrientationMaskPortrait;
}
*/

 -(BOOL)prefersStatusBarHidden{
 return YES;
 }
 

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setBorder];
    
    [self logOut:@"Initilalizing SDK..." withUITextView:logText withTimestamp:[self getTimestamp]];
    
    NSString *SDKVersion = [NSString stringWithFormat:@"%@ %@" , @"SDK Vesion", [[KidozSDK instance]getSdkVersion]];
    [self logOut:SDKVersion withUITextView:logText withTimestamp:[self getTimestamp]];
    

    [[KidozSDK instance]initializeWithPublisherID:@"8" securityToken:@"QVBIh5K3tr1AxO4A1d4ZWx1YAe5567os" withDelegate:self];
    
    NSString *soundFilePath = [NSString stringWithFormat:@"%@/WarmTutorialExplainer.mp3",[[NSBundle mainBundle] resourcePath]];
    NSURL *soundFileURL = [NSURL fileURLWithPath:soundFilePath];
    player = [[AVAudioPlayer alloc] initWithContentsOfURL:soundFileURL error:nil];
    player.numberOfLoops = -1;
    [player play];
    
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}




@end

File: KidozSDKSampleAppSwift/KidozSDKSampleAppSwift/ViewController.swift
//
//  ViewController.swift
//  Kidoz_iOS_SDK_Sample_App_Swift
//
//  Created by Lev Firer on 20/05/2020.
//  Copyright © 2020 Kidoz. All rights reserved.
//
import UIKit
class ViewController: UIViewController,KDZInitDelegate,KDZInterstitialDelegate,KDZRewardedDelegate,KDZBannerDelegate {
    
    var mBanner: UIView!
    @IBOutlet weak var logText: UITextView!
    
    override func viewDidLoad() {
          super.viewDidLoad()
          self.setBorder()
          self.initKidozSDK()
          // Do any additional setup after loading the view.
      }
    

    func setBorder(){
        self.logText.layer.borderColor = UIColor.orange.cgColor
        self.logText.layer.borderWidth = 2.3
        self.logText.layer.cornerRadius = 15
    }
    
    func logOut(message:String ,textView:UITextView ,timestamp:String){
        textView.text = textView.text + timestamp + " - " + message + "\n"
        let location = textView.text.count - 1
        let bottom = NSMakeRange(location, 1)
        textView.scrollRangeToVisible(bottom)
    }
    
    func getTimestamp () ->String {
        let formatter = DateFormatter()
        formatter.dateFormat = "HHmmssSSS"
        let time = formatter.string(from: Date())
        return time
    }
    
    
    // MARK: - SDK

   
    func initKidozSDK(){
        logOut(message: "Initilalizing SDK...",textView: logText,timestamp: getTimestamp())
        KidozSDK.instance().initialize(withPublisherID: "8", securityToken: "QVBIh5K3tr1AxO4A1d4ZWx1YAe5567os",with :self)
        logOut(message: "SDK Vesion " + KidozSDK.instance().getVersion(),textView: logText,timestamp: getTimestamp())
    }

    func onInitSuccess(){
        if(KidozSDK.instance().isSDKInitialized()){
            logOut(message: "SDK Initialized",textView: logText,timestamp: getTimestamp())
            NSLog("Init sdk success");
            
            logOut(message: "Interstitial initializing...",textView: logText,timestamp: getTimestamp())
            KidozSDK.instance().initializeInterstitial(with: self)

            logOut(message: "Rewarded initializing...",textView: logText,timestamp: getTimestamp())
            KidozSDK.instance().initializeRewarded(with: self)
            
            logOut(message: "Banner initializing...",textView: logText,timestamp: getTimestamp())
            KidozSDK.instance().initializeBanner(with: self,with: self)
            KidozSDK.instance().setBannerPosition(BOTTOM_CENTER)
        }
    };
    
    
    func onInitError(_ errorMessage: String!){
          NSLog("onInitError " + errorMessage);
          logOut(message: "onInitError " + errorMessage,textView: logText,timestamp: getTimestamp())
    }
    
    
    // MARK: - Interstitial
    
    @IBAction func loadInterstitial(_ sender: Any) {
        logOut(message: "loadInterstitial",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isSDKInitialized()){
            if(KidozSDK.instance().isInterstitialInitialized()){
                    KidozSDK.instance().loadInterstitial()
                }else{
                    logOut(message: "Interstitial not initialized",textView: logText,timestamp: getTimestamp())
                    NSLog("Interstitial not initialized");
                }
            }else{
            NSLog("sdk not initialized");
            logOut(message: "sdk not initialized",textView: logText,timestamp: getTimestamp())
        }
    }
    
    @IBAction func showInterstitial(_ sender: Any) {
        logOut(message: "showInterstitial",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isInterstitialReady()){
            KidozSDK.instance().showInterstitial()
        }else{
            NSLog("Interstitial not ready");
            logOut(message: "Interstitial not ready",textView: logText,timestamp: getTimestamp())
        }
    }
    
    func interstitialDidInitialize() {
        NSLog("interstitialDidInitialize");
        logOut(message: "interstitialDidInitialize",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialDidClose() {
        NSLog("interstitialDidClose");
        logOut(message: "interstitialDidClose",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialDidOpen() {
        NSLog("interstitialDidOpen");
        logOut(message: "interstitialDidOpen",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialIsReady() {
        NSLog("interstitialIsReady");
        logOut(message: "interstitialIsReady",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialReturnedWithNoOffers() {
        NSLog("interstitialReturnedWithNoOffers");
        logOut(message: "interstitialReturnedWithNoOffers",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialDidPause() {
        NSLog("interstitialDidPause");
        logOut(message: "interstitialDidPause",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialDidResume() {
        NSLog("interstitialDidResume");
        logOut(message: "interstitialDidResume",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialLoadFailed() {
        NSLog("interstitialLoadFailed");
        logOut(message: "interstitialLoadFailed",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialDidReciveError(_ errorMessage: String!) {
        NSLog("interstitialDidReciveError" + errorMessage);
        logOut(message: "interstitialDidReciveError",textView: logText,timestamp: getTimestamp())
    }
    
    func interstitialLeftApplication() {
        NSLog("interstitialLeftApplication");
        logOut(message: "interstitialLeftApplication",textView: logText,timestamp: getTimestamp())
    }
    
    // MARK: - Rewarded

    @IBAction func loadRewarded(_ sender: Any) {
        logOut(message: "loadRewarded",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isSDKInitialized()){
            if(KidozSDK.instance().isRewardedInitialized()){
                KidozSDK.instance().loadRewarded()
                }else{
                    logOut(message: "Rewarded not initialized",textView: logText,timestamp: getTimestamp())
                    NSLog("Rewarded not initialized");
                    }
        }else{
            NSLog("sdk not initialized");
            logOut(message: "sdk not initialized",textView: logText,timestamp: getTimestamp())
        }
    }
    
    @IBAction func showRewarded(_ sender: Any) {
        logOut(message: "showRewarded",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isRewardedReady()){
            KidozSDK.instance().showRewarded()
        }else{
            NSLog("Rewarded not ready");
            logOut(message: "Rewarded not ready",textView: logText,timestamp: getTimestamp())
        }
    }
    
    func rewardedDidInitialize() {
        NSLog("rewardedDidInitialize");
        logOut(message: "rewardedDidInitialize",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedDidClose() {
        NSLog("rewardedDidClose");
        logOut(message: "rewardedDidClose",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedDidOpen() {
        NSLog("rewardedDidOpen");
        logOut(message: "rewardedDidOpen",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedIsReady() {
        NSLog("rewardedIsReady");
        logOut(message: "rewardedIsReady",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedReturnedWithNoOffers() {
        NSLog("rewardedReturnedWithNoOffers");
        logOut(message: "rewardedReturnedWithNoOffers",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedDidPause() {
        NSLog("rewardedDidPause");
        logOut(message: "rewardedDidPause",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedDidResume() {
        NSLog("rewardedDidResume");
        logOut(message: "rewardedDidResume",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedLoadFailed() {
        NSLog("rewardedLoadFailed");
        logOut(message: "rewardedLoadFailed",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedDidReciveError(_ errorMessage: String!) {
        NSLog("rewardedDidReciveError" + errorMessage);
        logOut(message: "rewardedDidReciveError",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardReceived() {
        NSLog("rewardReceived");
        logOut(message: "rewardReceived",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedStarted() {
        NSLog("rewardedStarted");
        logOut(message: "rewardedStarted",textView: logText,timestamp: getTimestamp())
    }
    
    func rewardedLeftApplication() {
        NSLog("rewardedLeftApplication");
        logOut(message: "rewardedLeftApplication",textView: logText,timestamp: getTimestamp())
    }
    
    
    // MARK: - Banner
    
    @IBAction func loadBanner(_ sender: Any) {
        logOut(message: "loadBanner",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isSDKInitialized()){
            if(KidozSDK.instance().isBannerInitialized()){
                KidozSDK.instance().loadBanner()
                  }else{
                    logOut(message: "Banner not initialized",textView: logText,timestamp: getTimestamp())
                    NSLog("Banner not initialized");
                }
        }else{
            NSLog("sdk not initialized");
            logOut(message: "sdk not initialized",textView: logText,timestamp: getTimestamp())
        }
    }
    
    @IBAction func showBanner(_ sender: Any) {
        logOut(message: "showBanner",textView: logText,timestamp: getTimestamp())
        if(KidozSDK.instance().isBannerReady()){
            KidozSDK.instance().showBanner()
        }else{
            NSLog("Banner not ready");
            logOut(message: "Banner not ready",textView: logText,timestamp: getTimestamp())
        }
    }
    

    @IBAction func hideBanner(_ sender: Any) {
        KidozSDK.instance().hideBanner()
        logOut(message: "hideBanner",textView: logText,timestamp: getTimestamp())
    }
    
    
    func bannerDidInitialize() {
        NSLog("bannerDidInitialize");
        logOut(message: "bannerDidInitialize",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerDidClose() {
        NSLog("bannerDidClose");
        logOut(message: "bannerDidClose",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerDidOpen() {
        NSLog("bannerDidOpen");
        logOut(message: "bannerDidOpen",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerIsReady() {
        NSLog("bannerIsReady");
        logOut(message: "bannerIsReady",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerReturnedWithNoOffers() {
        NSLog("bannerReturnedWithNoOffers");
        logOut(message: "bannerReturnedWithNoOffers",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerLoadFailed() {
        NSLog("bannerLoadFailed");
        logOut(message: "bannerLoadFailed",textView: logText,timestamp: getTimestamp())
    }
    
    func bannerDidReciveError(_ errorMessage: String!) {
        NSLog("bannerLoadFailed" + errorMessage);
        logOut(message: "bannerLoadFailed" + errorMessage,textView: logText,timestamp: getTimestamp())
    }
    
    func bannerLeftApplication() {
        NSLog("bannerLeftApplication");
        logOut(message: "bannerLeftApplication",textView: logText,timestamp: getTimestamp())
    }
    
    func initBannerWithView(){
        let BANNER_WIDTH: CGFloat = 320
        let BANNER_HEIGHT: CGFloat = 50

        mBanner = UIView(frame: CGRect(x: 0, y: 0, width: BANNER_WIDTH, height: BANNER_HEIGHT))
        view.addSubview(mBanner)
        mBanner.translatesAutoresizingMaskIntoConstraints = false
        view.addConstraints([NSLayoutConstraint(
                    item: mBanner,
                    attribute: .bottom,
                    relatedBy: .equal,
                    toItem: bottomLayoutGuide,
                    attribute: .top,
                    multiplier: 1,
                    constant: 0),
                NSLayoutConstraint(
                    item: mBanner,
                    attribute: .centerX,
                    relatedBy: .equal,
                    toItem: view,
                    attribute: .centerX,
                    multiplier: 1,
                    constant: 0),
                NSLayoutConstraint(
                    item: mBanner,
                    attribute: .height,
                    relatedBy: .equal,
                    toItem: nil,
                    attribute: .notAnAttribute,
                    multiplier: 0,
                    constant: BANNER_HEIGHT),
                NSLayoutConstraint(
                    item: mBanner,
                    attribute: .width,
                    relatedBy: .equal,
                    toItem: nil,
                    attribute: .notAnAttribute,
                    multiplier: 0,
                    constant: BANNER_WIDTH)])
        
                view.bringSubviewToFront(mBanner)
                KidozSDK.instance().initializeBanner(with: self, with: mBanner)
        
    }
    
  
}

